Fix timestamp formatting and handle entries with missing timestamps

leec3108 opened the pull request:
These commits add code to fix the timestamp formatting (which brings it in line with the Euro-style decimals many third-party utilities expect to see), and to handle a case with some transcriptions where the previous entry (usually a punctuation entry) has no timestamps.

leec3108 committed:
- Fix the time stamp formatting. Many utilities won't accept the format unless the decimal uses a comma (European-style).
- We found in our testing on real files that some entries from Transcription don't have timestamps. Instead of blindly trusting the
  previous entry, roll back through the list until we find an entry with a valid end time we can use.

leec3108 committed:
Update the test file to reflect the formatting fixes.

printscreen commented on index.js:
```diff
@@ -1,5 +1,13 @@
+	
+    return json.results.items[index].end_time;
+}
+
```
Can you convert the tabs to spaces as the rest of the project uses spaces?

Also, is there a possibility that this loop could never end and get stuck?

leec3108 replied:
In my testing I didn't have a problem with the loop getting stuck, but thinking about it, it shouldn't be going back further than two or three items to find a good timestamp. I'll check it out.

leec3108 replied:
I'm rewriting getPreviousEndTime to limit the number of items it looks at. What do you recommend it doing if it can't find an item to use?

leec3108 replied:
This sounds good to me.

leec3108 committed:
- Convert tabs to spaces.
- Rewrite getPreviousEndTime so it will stop and throw an error if it can't find a valid timestamp within 2 or 3 items.

leec3108 committed:
Lambda's editor was upset about previousEndTime not being defined, so define it now.

leec3108 commented:
Just uploaded some new commits that resolve the issues mentions. Please take a look and let me know what you think.

leec3108 committed:
Remove the stopAt functionality. Not overrunning the end of the list is more important than being quick here, and this code is simpler and easier to follow.

leec3108 commented:
New commit. I simplified the way getPreviousEndTime works, since it's more important that the lookup stops at index 0 if has to, than it is for it to stop at a certain distance.
